src/test/v2.0/advanced/open-support-2.0.ts

```diff
@@ -0,0 +1,205 @@
+import { assert, expect } from "chai";
+import {
+  Channel,
+  Context,
+  DesktopAgent,
+  getOrCreateChannel,
+  Listener,
+  OpenError,
+} from "fdc3_2_0";
+import APIDocumentation from "../../../apiDocuments";
+import constants from "../../../constants";
+import { sleep, wait, wrapPromise } from "../../../utils";
+import { AppControlContext } from "../../common/channel-control";
+import { MockAppContext, OpenControl } from "../../common/open-control";
+
+declare let fdc3: DesktopAgent;
+const openDocs = "\r\nDocumentation: " + APIDocumentation.open + "\r\nCause:";
+const testTimeoutMessage = `Test timeout - An error was not thrown within the allocated timeout of ${constants.NoListenerTimeout}. This timeout is not defined by the standard, rather by each implementation. Hence, if you DA implementation uses a longer timeout the constants.NoListenerTimeout in the test framework will need to be increased.`;
+let timeout;
+
+export class OpenControl2_0 implements OpenControl<Context> {
+  contextReceiver = async (contextType: string): Promise<Context> => {
+    const appControlChannel = await getOrCreateChannel("app-control");
+    let timeout;
```

Comment:
With regard to it being scoped inside each test, is that because the rejection with the error message will not show in the test?

I have found that when rejecting with an error like this within the `contextReceiver` function: `reject(new Error("No context received from app B"));` The error message does show up in the test itself without having to check the console for it

Reply:
Also, I can return it from the util function, but I would have to lock it to use fdc3 API v2. This function is used by v1 and v2 tests. It shouldn't make any difference in theory as the only difference is that getOrCreateChannel returns a promise in v2, but it would be nice to keep v1 and v2 stuff separate for consistency

Reply:
>Also, I can return it from the util , but I would have to lock it to use fdc3 API v2

Why would you have to lock it?

>With regard to it needing to be scoped inside each test, is that because the rejection with the error message will not show in the test?

No, its just generally bad form to be using and mutating variables in the global scope when they can be locally scoped (so this is mostly a style comment). Whilst I don't see it producing incorrect behaviour in the tests currently it may trip someone up in the future - particularly if you end up needing two timeouts in a single test (which could totally happen). As the timeout is set and released in the local scope you _should_ keep the variable there. 

A common, but hard to debug, issue you might run into in the future is working with a variable called timeout somewhere else also in the global scope, it would then overwrite this one, you end up clearing the wrong one here etc..

Reply:
Ah, you mean the global `timeout` variable needs to be scoped. That makes sense. The highlighted code is not showing the global `timeout` variable, hence the confusion. Disregard everything I said

Reply:
Bad Github - shows correctly on files tab: 
![image](https://user-images.githubusercontent.com/1701764/204822874-6134ac0d-b6e4-4024-ab5f-c6157ffa66ab.png)

Reply:
This is what I see 
![image](https://user-images.githubusercontent.com/107405201/204823303-1fd8cd6a-49f9-426b-8ec4-c3aa1cdc7d7a.png)

Reply:
I've addressed all the comments raised. This can now be merged if you are happy with the changes. However... just to throw a spanner in the works, I've opened another PR that builds on this and that is more of a suggestion. If this other PR requires more time to think about/review we can merge this in first and deal with that one at a later date:

https://github.com/finos/FDC3-conformance-framework/pull/145